`lib/vfscore`: Build warning: `warning: ‘struct dirent64’ declared inside parameter list will not be visible outside of this definition or declaration`

### Describe the bug

Building applications using `vfscore` results in a build warning.

### Steps to reproduce

Build Nginx for QEMU/x86_64 in the [`catalog-core` repository](https://github.com/unikraft/catalog-core). Follow the steps:

1. Clone the repository:

   ```console
   git clone https://github.com/unikraft/catalog-core
   ```

1. Enter the `nginx/` directory:

   ```console
   cd catalog-core/nginx/
   ```
   
1. Switch to the `scripts` branch:

   ```console
   git checkout scripts
   ```

1. Build Nginx for QEMU/x86_64:

   ```console
   ./build.qemu.x86_64
   ```

   The build warning is printed.

### Expected behavior

Build runs without warning.

### Which architectures were you using or does this bug affect?

_No response_

### Which platforms were you using or does this bug affect?

_No response_

### Relevant log output

```shell
In file included from /home/razvan/orgs/unikraft/tmp/test-nginx-core/catalog-core/repos/unikraft/lib/posix-fdio/fdctl.c:18:                                                                   
/home/razvan/orgs/unikraft/tmp/test-nginx-core/catalog-core/repos/unikraft/lib/vfscore/include/vfscore/vnode.h:175:41: warning: ‘struct dirent64’ declared inside parameter list will not be visible outside of this definition or declaration                                                                                                                                              
  175 |                                  struct dirent64 *);
```

--- a/lib/posix-fdio/fdctl.c
+++ b/lib/posix-fdio/fdctl.c
@@ -6,6 +6,10 @@
 
 /* Internal syscalls for file control operations */
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #include <sys/ioctl.h>
 
 #include <uk/atomic.h>

--- a/lib/posix-poll/epoll.c
+++ b/lib/posix-poll/epoll.c
@@ -4,6 +4,10 @@
  * You may not use this file except in compliance with the License.
  */
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #include <errno.h>
 
 #include <uk/atomic.h>

--- a/lib/ramfs/ramfs_vfsops.c
+++ b/lib/ramfs/ramfs_vfsops.c
@@ -30,6 +30,10 @@
  * SUCH DAMAGE.
  */
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #include <errno.h>
 
 #define _BSD_SOURCE

--- a/lib/vfscore/automount.c
+++ b/lib/vfscore/automount.c
@@ -37,6 +37,11 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
+
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #include <stdlib.h>
 #include <string.h>
 #include <inttypes.h>

